cli/popper/gha.py

```diff
@@ -53,6 +55,61 @@ def __init__(self, wfile, workspace):
         for e in dict(self.env):
             self.env.update({e.replace('GITHUB_', 'POPPER_'): self.env[e]})
 
+    def validate(self):
+        """
+        Validates the .workflow file
+        """
+        if not self.wf.get('workflow', None):
+            pu.fail('Workflow block must be present\n')
+        for _, wf_block in dict(self.wf['workflow']).items():
+            if not wf_block.get('resolves', None):
+                pu.fail('[resolves] attribute must be present\n')
+            else:
+                if not isinstance(
```

Comment:
It might be cleaner to do this check in the `normalize()` function, since we are doing type checking there already